components/form-builder/hooks/useTemplateStatus.tsx

```diff
@@ -0,0 +1,54 @@
+import { useEffect, useState, useCallback } from "react";
+import { useSession } from "next-auth/react";
+import axios from "axios";
+import { logMessage } from "@lib/logger";
+
+import { useTemplateStore } from "../store";
+
+interface FormTemplate {
+  id: string;
+  updatedAt: number;
+}
+
+export const byId = async (id: string): Promise<FormTemplate | null> => {
+  try {
+    const result = await axios({
+      url: "/api/templates",
+      method: "GET",
+      headers: { "Content-Type": "application/json" },
+      data: { formID: id },
+      timeout: process.env.NODE_ENV === "production" ? 60000 : 0,
+    });
+
+    if (result.status !== 200) {
+      return null;
+    }
+
+    return result?.data.find((template: FormTemplate) => template.id === id);
+  } catch (err) {
+    logMessage.error(err);
+    return null;
+  }
+};
+
+export const useTemplateStatus = () => {
```

Comment:
Moved this call to the API from the save button to a hook --- for easier re-use if needed later somewhere else.